src/api/sfc-script-setup.md

```diff
@@ -227,6 +227,77 @@ const props = withDefaults(defineProps<Props>(), {
 
 This will be compiled to equivalent runtime props `default` options. In addition, the `withDefaults` helper provides type checks for the default values, and ensures the returned `props` type has the optional flags removed for properties that do have default values declared.
 
+## defineModel() <sup class="vt-badge" data-text="3.4+" /> {#definemodel}
+
+This macro can be used to declare a two-way binding prop that can be consumed via `v-model` from the parent component. Example usage is also discussed in the [Component `v-model`](/guide/components/v-model) guide.
+
+Under the hood, this macro declares a model prop and a corresponding value update event. If the first argument is a literal string, it will be used as the prop name; Otherwise the prop name will default to `"modelValue"`. In both cases, you can also pass an additional object which can include the prop's options and the model ref's value transform options.
+
+```js
+// declares "modelValue" prop, consumed by parent via v-model
+const model = defineModel()
+// OR: declares "modelValue" prop with options
+const model = defineModel({ type: String })
+
+// emits "update:modelValue" when mutated
+model.value = 'hello'
+
+// declares "count" prop, consumed by parent via v-model:count
+const count = defineModel('count')
```

Comment:
_some random thoughts_

Is it possible to declare multiple models like that?
```js
// v-model:count
const count = defineModel('count')
// v-model:x
const x = defineModel('x')
```
Or will this result in an error?
What happens if someone accidentally redeclared two times the same?
```js
const count = defineModel('count')
const count2 = defineModel('count')
```
Should this be handled by a linter? Will there be a Vue runtime error?

Reply:
First case is supported. Second case will throw a compile-time error.

Reply:
Could we add docs that different `defineModel`s are supported and can be used that way? I think it is really common to have multiple `v-model:*` but it works differently as to `defineProps` or `defineEmits` where you pass multiple values at once, but for `defineModel` you _suddenly_ need multiple calls.

Reply:
It is already shown in the component v-model guide.